pkg/agent/agent.go

```
@@ -279,57 +276,31 @@ func (k *klocksmith) process(ctx context.Context) error {
 		klog.Info("Node already marked as unschedulable")
 	}
 
-	klog.Info("Getting pod list for deletion")
-
-	pods, err := k.getPodsForDeletion(ctx)
-	if err != nil {
-		return fmt.Errorf("getting list of pods for deletion: %w", err)
-	}
-
-	// Delete the pods.
-	// TODO(mischief): Explicitly don't terminate self? we'll probably just be a
-	// Mirror pod or daemonset anyway..
-	klog.Infof("Deleting %d pods", len(pods))
-
-	for _, pod := range pods {
-		klog.Infof("Terminating pod %q...", pod.Name)
-
-		if err := k.pg.Pods(pod.Namespace).Delete(ctx, pod.Name, metav1.DeleteOptions{}); err != nil {
-			// Continue anyways, the reboot should terminate it.
-			klog.Errorf("Failed terminating pod %q: %v", pod.Name, err)
-		}
+	dh := drain.Helper{
+		Ctx:                 ctx,
```

We should be also specifying `Timeout: k.reapTimeout` here perhaps? Or should this value be used in `GracePeriodSeconds`?

---

`Timeout` isn't documented but looks to be a global timeout for the drain itself, when reached it will return an error
`GracePeriodSeconds` is documented as follows
```	
// GracePeriodSeconds is how long to wait for a pod to terminate.
// IMPORTANT: 0 means "delete immediately"; set to a negative value
// to use the pod's terminationGracePeriodSeconds.
```
`reapTimeout` was implemented as a per pod delete timeout at which point the wait/poll func would return (not sure with with an error or not) but in either scenario log the pod name and return

GracePeriodSeconds is probably closest in terms of behaviour but i'd like to change the default value of `reapTimeout` to -1 which will cause it to base the per pod termination timeouts based on the pods `TerminationGracePeriod` if configured, or the kubernetes default of 30s

---

Hmm, so in existing implementation, we wait `reapTimeout` per pod, while drain helper has a global timeout for all pods, so the window will be slightly smaller I guess, though I suppose it should still be acceptable.

Given that we ignore all errors from deleting pods anyway and we only log them, I think it should be fine to set the `Timeout` field to `reapTimout`. And `reapTimeout` is also user configurable, so if there is a regression in waiting time, user can still increase it if needed.

Maybe I'll open an issue to implement some smarter way to respect `TerminationGracePeriod`.

---

Thank you! I'll review the PR early next week.